executive_logic/executive_logic.py

```diff
@@ -206,40 +213,35 @@ def store_query(self, command: MessageType, args: list):
         """
         self.query_response = Message(command, args)  # Store server response in self.query_response
 
-    def __update_scores_tokens_spins(self):
+    def __update_scores_tokens_spins(self, curr_player_id):
         """
         Called after a game value is updated that should be reflected in the UI display.
         Sends server the current player scores, player tokens, and number of spins remaining.
         :return: void
         """
-        self.__query_server(MessageType.UPDATE_SCORES,
+        self.__query_server(curr_player_id, MessageType.UPDATE_SCORES,
```

Comment:
Are you only notifying the current player of the updated score? I think the scoreboard should be updated for all the players still.

Reply:
I'm seeing now that update_scores calls __query_server which waits for a response from the client. It probably makes sense for the "update" type messages to call a different function that just pushes the updated and doesn't make the server wait for a response from the client.s